src/gpuarray_buffer_cuda.c

```diff
@@ -147,13 +148,49 @@ static int setup_lib(void) {
         return error_set(global_err, GA_IMPL_ERROR, "cuDriverGetVersion failed");
       major = tmp / 1000;
       minor = (tmp / 10) % 10;
+      #if defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64) || defined(__APPLE__)
+      /* We will dynamically search the right CUDA version only on Windows and Macintosh systems,
+      and only if user has not explicitely specified GPUARRAY_CUDA_VERSION. */
+      search_version = 1;
+      #endif
     } else {
       major = ver[0] - '0';
       minor = ver[1] - '0';
     }
+    /* NB: next line will cause problems if a CUDA 10.0 (or 9.11) is released in the future. */
     if (major > 9 || major < 0 || minor > 9 || minor < 0)
       return error_fmt(global_err, GA_VALUE_ERROR, "Invalid cuda version: %d.%d", major, minor);
-    res = load_libnvrtc(major, minor, global_err);
+    if (!search_version) {
+      res = load_libnvrtc(major, minor, global_err);
+    } else {
+      /* First case in next array is reserved to eventually receive the version returned by cuDriverGetVersion(). */
+      int versions[] = {-1, 80, 75};
+      int versions_length = sizeof(versions) / sizeof(int);
+      int current_version = major * 10 + minor;
+      int i = 0;
+      for (i = 1; i < versions_length && versions[i] != current_version; ++i);
+      if (i == versions_length) {
+        /* Current version not found in the list of versions. We add it at top of the list. */
+        versions[0] = current_version;
+        /* We will iterate on versions from the first. */
+        i = 0;
+      } else {
+        /* Current version found in the list of known versions. No need to add it to the list. */
+        i = 1;
+      };
```

Comment:
I did that to avoid testing twice a same version that is not installed (e.g. if cuDriverGetVersion() returns 80, then 80 will already be in the list of supported versions).

Do you anyway want the fallback strategy ?